tests/ut/ir/transforms/test_l0_tile_chooser.py

```
@@ -354,6 +357,86 @@ def test_invalid_negative_dim_raises(self):
 
+    def test_optional_max_n_caps_the_legal_tile_grid(self):
+        """Auxiliary N-only SRAM constraints restrict candidates without changing the cost model."""
+        cfg = _default_config(M=256, N=1024, K=128)
+        cfg.max_n = 64
+        result = passes.l0_tile_chooser.choose_l0_tile(cfg)
+        assert result.n <= 64
+
+    def test_max_n_must_fit_the_minimum_tile(self):
+        cfg = _default_config(M=128, N=128, K=128)
+        cfg.max_n = 8
+        with pytest.raises(ValueError, match="max_n must be zero .* or at least min_n"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)